app/src/main/java/space/taran/arkretouch/data/ImageDefaults.kt

```diff
@@ -0,0 +1,34 @@
+package space.taran.arkretouch.data
+
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.IntSize
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+@Serializable
+data class ImageDefaults(
+    val colorValue: ULong = Color.White.value,
+    val resolution: Resolution? = null
+) {
+    override fun toString() = Json.encodeToString(this)
+
+    companion object {
+        fun fromString(string: String): ImageDefaults {
+            return Json.decodeFromString(string)
+        }
+    }
+}
```

Comment:
No need to override `toString` and `fromString` with json
You can do mapping directly in Preferences